File: FunctionTestRunner/FunctionTestRunner/Utils/Wait.cs
namespace FunctionTestRunner.Utils;

public static class Wait
{
    public static void ForMilliseconds(double milliseconds)
    {
        WaitInternal(TimeSpan.FromMilliseconds(milliseconds));
    }

    public static void ForSeconds(double seconds)
    {
        WaitInternal(TimeSpan.FromSeconds(seconds));
    }

    public static void ForMinutes(double minutes)
    {
        WaitInternal(TimeSpan.FromMinutes(minutes));
    }

    private static void WaitInternal(TimeSpan timespan)
    {
        Thread.Sleep(timespan);
    }
}

File: FunctionTestRunnerTS/FunctionTestRunner/src/utils/wait.ts
export default class Wait {
    static async forMilliseconds(milliseconds: number)
    {
        await this.waitInternal(milliseconds);
    }

    static async forSeconds(seconds: number)
    {
        await this.waitInternal(seconds * 1000);
    }

    static async forMinutes(minutes: number)
    {
        await this.waitInternal((minutes * 60) * 1000);
    }

    private static async waitInternal(milliseconds: number)
    {
        return new Promise(resolve => setTimeout(resolve, milliseconds));
    }
}